test/parallel/test-https-agent-sockets-leak.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+const common = require('../common');
+if (!common.hasCrypto) {
+  common.skip('missing crypto');
+  return;
+}
+
+const fs = require('fs');
+const https = require('https');
+const assert = require('assert');
+
+const options = {
+  key: fs.readFileSync(common.fixturesDir + '/keys/agent1-key.pem'),
+  cert: fs.readFileSync(common.fixturesDir + '/keys/agent1-cert.pem'),
+  ca: fs.readFileSync(common.fixturesDir + '/keys/ca1-cert.pem')
+};
+
+const server = https.Server(options, common.mustCall((req, res) => {
+  res.writeHead(200);
+  res.end('https\n');
+}));
+
+const agent = new https.Agent({
+  keepAlive: false
+});
+
+server.listen(0, common.mustCall(() => {
+  https.get({
+    host: server.address().host,
+    port: server.address().port,
+    headers: {host: 'agent1'},
+    rejectUnauthorized: true,
+    ca: options.ca,
+    agent: agent
+  }, common.mustCall((res) => {
+    res.resume();
+    server.close();
+
+    // Only one entry should exist in agent.sockets pool
+    // If there are more entries in agent.sockets,
+    // removeSocket will not delete them resulting in a resource leak
+    assert.strictEqual(Object.keys(agent.sockets).length, 1);
+
+    res.req.on('close', common.mustCall(() => {
+      // To verify that no leaks occur, check that no entries
+      // exist in agent.sockets pool after both request and socket
+      // has been closed.
+
```

Comment:
I think this line should also be removed? https://github.com/nodejs/node/pull/8647#discussion_r79305039